templates/ehub.json

```diff
@@ -198,23 +234,25 @@
             ]
         },
         {
+            "comments": "Assign contributor role to the app principal id for the resource group application is deployed to.",
             "type": "Microsoft.Authorization/roleAssignments",
-            "name": "[variables('roleAssignmentId')]",
+            "name": "[variables('contribRoleAssignmentName')]",
             "apiVersion": "2015-07-01",
             "properties": {
-                "roleDefinitionId": "[variables('contributor')]",
+                "roleDefinitionId": "[variables('contributorRole')]",
                 "principalId": "[parameters('Service Principal ID')]",
                 "scope": "[variables('resourceGroupId')]"
             }
         },
         {
+            "comments": "Create new event hub namespace and event hub in case of new deployment.",
```

Comment:
Does this create the ehub only if it doesn't exist?

Reply:
I missed a condition here.